src/app/libp2p_helper/src/codanet.go

```diff
@@ -514,6 +517,28 @@ func (h *Helper) handlePxStreams(s network.Stream) {
 	}
 }
 
+func (h *Helper) handleTelemetryStreams(s network.Stream) {
+	defer func() {
+		_ = s.Close()
+	}()
+	bz, err := json.Marshal(h.TelemetryData)
```

Comment:
Does it need to be JSON / would it just be cheaper to somehow write the string directly?

Reply:
the JSON only adds two bytes, the marshalled string `"testdata"` is this: `[34 116 101 115 116 100 97 116 97 34]` it's just easier to code cause it deals w/ the length for you, but would work either way. lemme know if you want me to change it